src/analysis/event_detector.py

```diff
@@ -0,0 +1,312 @@
+import xarray as xr
+import numpy as np
+from pathlib import Path
+from xclim.run_length import rle, longest_run  # , windowed_run_events
+from typing import Tuple, Optional, Any
+import warnings
+import pandas as pd
+from scripts.eng_utils import get_ds_mask
+
+
+class EventDetector():
+    def __init__(self,
+                 path_to_data: Path,
+                 data_folder: Path = Path('data')) -> None:
+        self.data_folder = data_folder
+        self.interim_folder = data_folder / "interim"
+        self.processed_folder = data_folder / "processed"
+
+        if not self.processed_folder.exists():
+            self.processed_folder.mkdir()
+
+        assert path_to_data.exists(), f"{path_to_data} does not point to an existing file!"
+        self.ds = self.read_data(path_to_data)
+
+    def read_data(self, path_to_data: Path) -> xr.Dataset:
+
+        try:
+            ds = xr.open_dataset(path_to_data)
+
+        except ValueError:
+            print(ValueError)
+            warnings.warn("Having to decode_times=False because unsupported calendar")
+            ds = xr.open_dataset(path_to_data, decode_times=False)
+            warnings.warn("Hardcoding MONTHLY data (CHIRPS example)")
+            time = pd.date_range(start='1900-01-01', freq='M', periods=len(ds.time.values))
+            ds['time'] = time
+
+        print(f"{path_to_data.name} read!")
+        ds = ds.sortby('time')
+        # TODO: infer time frequency ?
+        return ds
+
+    @staticmethod
+    def calculate_threshold(ds: xr.Dataset,
+                            clim: xr.Dataset,
+                            method: str,
+                            time_period: str,
+                            variable: str,
+                            hilo: Optional[str] = None,
+                            value: Optional[float] = None,) -> xr.DataArray:
+        """Calculate the threshold based on the `method` argument
+        method: str
+            ["q90","q10","std","abs",]
```

Comment:
What do these different methods do?
Can an explanation be included in the docstring?

Reply:
(its kind of self explanatory looking at the code, but a short description in the docstring would still be helpful)

Reply:
thanks!